### PR TITLE
Show the number of unread notifications above the bell on the right

### PR DESCRIPTION
**Pull request by turt2live**
~~Requires https://github.com/matrix-org/matrix-js-sdk/pull/802~~
Fixes https://github.com/vector-im/riot-web/issues/3383

This achieves the result by counting up the number of highlights across all rooms and setting that as the badge above the icon. If there are no highlights, nothing is displayed. The red highlight on the bell is done by abusing how the Tinter works: because it has access to the properties of the SVG that we'd need to override it, we give it a collection of colors it should use instead of the theme/tint it is trying to apply. This results in the Tinter using our warning color instead of whatever it was going to apply.

The RightPanel now listens for events to update the count too, otherwise when the user receives a ping they'd have to switch rooms to see the change.

image
Darker color is me testing tinting.

### CODE REVIEW

**Comment by turt2live** [comment]
This has a bug where when reading (or receiving new) notifications the icon doesn't switch colors and remains stuck.

File: src/components/structures/RightPanel.js [hunk 1]
```diff
@@ -205,7 +216,10 @@ module.exports = React.createClass({
     }, 500),
 
     onAction: function(payload) {
-        if (payload.action === "view_user") {
+        if (payload.action === "event_notification") {
+            // Try and re-caclulate any badge counts we might have
+            this.forceUpdate();
```
Comment (dbkr) [inline src/components/structures/RightPanel.js]: Will you not need an equivalent for when the user reads the notification and the count goes down again?

Comment (turt2live) [reply src/components/structures/RightPanel.js]: You would think so, but something somewhere magically makes it work. I tested in Chrome and Firefox and for some reason it happily goes down but doesn't magically go up.

Comment (dbkr) [reply src/components/structures/RightPanel.js]: yay magic :/

**Pull request merged**